gpflow/base.py

```diff
@@ -74,7 +64,7 @@ def __init__(
 
         self._transform = transform
         self.prior = prior
-        self.prior_on = prior_on
+        self._prior_on = prior_on
```

Comment:
this is changing behaviour, it will break passing in a string as `prior_on`.

Reply:
are you sure? we have the getters and setters defined below (which is why I made this change)

Reply:
Try it out for yourself! If the `__init__` is called with, e.g., `prior_on="unconstrained"`, then this line assigns `self._prior_on` to be that string, bypassing the setter - which would have transformed it from a string to the corresponding PriorOn enum instance.

Reply:
ah, i didn't realise that this would call the setter. i saw that `_prior_on` isn't defined on explicitly in `__init__`. i wonder if it might be safer to use `self._prior_on = Prior(prior_on)` in `__init__`. Either way, it's not needed in this PR, so I'll change it back